src/lib/containers/download_list/DownloadConfirmation.tsx

```diff
@@ -219,7 +207,7 @@ export const DownloadConfirmation: React.FunctionComponent<DownloadConfirmationP
     if (queryResultBundle) {
       updateStats(queryResultBundle.queryCount, queryResultBundle.sumFileSizes?.sumFileSizesBytes)
     }
-  }, [updateStats, queryOrFolder, queryResultBundle])
+  }, [updateStats, lastQueryRequest, queryResultBundle])
```

Comment:
And this effect is specific to the query request (get the stats on the rowset that this query would return)